pyasic/miners/backends/vnish.py

```diff
@@ -270,3 +271,19 @@ async def get_config(self) -> MinerConfig:
             return self.config
         self.config = MinerConfig.from_vnish(web_settings)
         return self.config
+
+    async def set_power_limit(self, wattage: int) -> bool:
+        # Can only set power limit to tuned preset
+        try:
+            await self.web.set_power_limit(wattage)
+            updated_settings = await self.web.settings()
+        except APIError:
+            raise
+        except Exception as e:
+            logging.warning(f"{self} - Failed to set power limit: {e}")
+            return False
+
+        if updated_settings["miner"]["overclock"]["preset"] == str(wattage):
```

Comment:
This conversion should probably be to int for the preset rather than to str for the wattage.

Reply:
I revised this in [10b1760](https://github.com/UpstreamData/pyasic/pull/245/commits/10b17608c429cad75862422dc70d13c237ce2709)